src/environment.ts

```diff
@@ -148,14 +148,19 @@ export class Environment {
     const { dataVarname, autoDataVarname } = this.options;
 
     if (autoDataVarname) {
-      try {
-        code = transformTemplateCode(code, dataVarname);
-      } catch (cause) {
-        if (cause instanceof TransformError) {
-          throw new TemplateError(path, source, cause.position, cause);
-        }
-
-        throw new Error(`Unknown error while transforming ${path}`, { cause });
+      const generator = iterateTopLevel(code);
+      const [, , variables] = generator.next().value;
+      while (!generator.next().done);
+      variables.delete(dataVarname);
+      if (variables.size > 0) {
+        code = `
+          var ${
+          [...variables].map((name) => {
+            return `${name} = ${dataVarname}.${name} ?? globalThis.${name}`;
```

Comment:
This is ugly, but variables need to fall back to `globalThis` and the spread operator only spreads enumerable properties. Without this, e.g. `Promise` becomes `undefined`. The alternative is wrapping `it` with a `Proxy` that returns either the property on the `it` if it exists or the global variable otherwise. Probably that's neater overall, but curious if you have any ideas for this.